### PR TITLE
feat: enables intree builds for mlir-xten

### PR DESCRIPTION
**Pull request by maxbartel**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,14 +24,11 @@ if(POLICY CMP0116)
   cmake_policy(SET CMP0116 OLD)
 endif()
 
-project(pytorch_mlir LANGUAGES CXX C)
```
Comment (gargaroff) [inline CMakeLists.txt:28]: I am surprised this has been wrong the whole time

File: CMakeLists.txt [hunk 2]
```diff
@@ -77,32 +99,29 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
...
 
 add_definitions(${LLVM_DEFINITIONS})
 
-add_custom_target(check-all)
+add_custom_target(check-xten-all)
 
 # Make sure we build the docs
-add_custom_target(docs ALL)
-add_dependencies(docs mlir-doc)
+if(${LLVM_BUILD_DOCS})
```
Comment (gargaroff) [inline CMakeLists.txt]: This should only be checked if we are in an in-tree build

**Review by gargaroff — APPROVED** [review]

**Review by ljfitz — APPROVED** [review]
Thanks Max.

**Comment by gargaroff** [comment]
Should we merge this? I don't see a reason for keeping this open

**Comment by gargaroff** [comment]
- [x] on hold until Xilinx forks are ready

**Comment by maxbartel** [comment]
@mgehre-amd Was this still working?

**Comment by mgehre-amd** [comment]
Didn't seem to break anything when I tested locally

**Pull request merged**